src/typechecker.rs

```diff
@@ -54,8 +54,10 @@ pub fn can_fit_integer(type_id: TypeId, value: &IntegerConstant) -> bool {
     match *value {
         IntegerConstant::Signed(value) => match type_id {
             I8_TYPE_ID => value >= i8::MIN as i64 && value <= i8::MAX as i64,
+            CCHAR_TYPE_ID => value >= i8::MIN as i64 && value <= i8::MAX as i64,
             I16_TYPE_ID => value >= i16::MIN as i64 && value <= i16::MAX as i64,
             I32_TYPE_ID => value >= i32::MIN as i64 && value <= i32::MAX as i64,
+            CINT_TYPE_ID => value >= i32::MIN as i64 && value <= i32::MAX as i64,
```

Comment:
- Let's put these at the top and next to each other, like in `is_integer
- Pretty sure we should be using `https://doc.rust-lang.org/std/os/raw/type.c_int.html` (and `c_char`) for MIN and MAX here, they're not guaranteed to be i8 and i32 on all platforms and architectures